apps/sms/test/marionette/mocks/mock_split_view_mode.js

```diff
@@ -0,0 +1,14 @@
+/* global Services */
+'use strict';
+
+Services.obs.addObserver(function(document) {
+  if (!document || !document.location ||
+      document.location.origin !== 'app://sms.gaiamobile.org') {
+    return;
+  }
+
+  // Switch to split-view mode Inbox instead.
+  if(document.location.pathname === '/index.html') {
```

Comment:
Mmm, what do you mean by synchronous function? If you mean that here we guarantee that application will not start before URL is changed - then I believe yes - we call ```replace```, that will fire all ```document-element-inserted``` observers once again before app is actually started (including this one).